libeos-update-server/server.c

```diff
@@ -255,15 +255,17 @@ update_pending_requests (EusServer *self,
   GObject *obj = G_OBJECT (self);
 
   g_assert (delta >= 0 || self->pending_requests >= (guint) -delta);
-  g_assert (delta <= 0 || self->pending_requests <= G_MAXUINT - delta);
+  g_assert (delta <= 0 || self->pending_requests <= G_MAXUINT - (guint) delta);
 
   if (delta == 0)
     return;
 
-  g_debug ("%s: Updating from %u to %u", G_STRFUNC, self->pending_requests,
-           self->pending_requests + delta);
+  g_debug ("%s: Updating from %u by %d", G_STRFUNC, self->pending_requests, delta);
 
-  self->pending_requests += delta;
+  /* Silence a -Wconversion warning. @delta may be negative, in which case the
+   * beauty of two’s-complement will give us the result we’d expect if the
+   * (guint) cast was not explicit. */
```

Comment:
I'm surprised we're relying on technically undefined behavior but I guess if that's how practically every compiler and architecture works it's fine.

Reply:
Ooops, you’re right, C doesn’t technically have to be implemented using two’s-complement. I’ll rework the commit to not rely on that. This is getting ridiculous though.

Reply:
Reworked in commit dc692c96cbd0a2c3e2ebafa3b6b0676a58f201bd.